packages/nextjs/app/debug/_components/contract/WriteOnlyFunctionForm.tsx

```diff
@@ -41,17 +44,31 @@ export const WriteOnlyFunctionForm = ({
 
   const { data: result, isPending, writeContractAsync } = useWriteContract();
 
+  const wagmiConfig = useConfig();
+
+  const simulateAndNotifyError = async (params: WriteContractVariables<Abi, string, any[], Config, number>) => {
```

Comment:
maybe add `wagmiConfig` as param an move to utils or keep in one file, since the function is duplicated?

Reply:
Ohh yeah initially I thought since its being used in just two files so not worth it, but yeah makes sense to have it in util 🙌

Just moved it under `utils/scaffold-eth/contract.ts` lol but not 100% sure if it's the right place since that file contains all types but also makes sense to have that function in that file since its util function is related to contract interaction.

Reply:
Thanks! Probably we need to move types to some separated types file(s) later

Reply:
Ohh yes 💯!! Has been in my mind always!